lib/Common/Common/Int32Math.cpp

```diff
@@ -176,17 +176,31 @@ Int32Math::Not(int32 val, int32 *pResult)
 bool
 Int32Math::Inc(int32 val, int32 *pResult)
 {
+#ifdef _MSC_VER
     *pResult = val + 1;
     // Overflow if result ends up less than input
     return *pResult <= val;
+#elif defined(__APPLE__)
+    *pResult = val + 1;
+    return val == INT32_MAX; // Overflow if val was int max
+#else
+    return __builtin_add_overflow(val, 1, pResult);
```

Comment:
Actually AppleClang has that. Don't know why CI complained.

Instead of making it `ifdef __APPLE__` we better feature detect from clang.

```
#ifdef __has_builtin
#if __has_builtin(__builtin_add_overflow)
```
Same applies to other builtin comments below